docs/use_cases/hybrid_search_&_rerank_rag.md

```diff
@@ -0,0 +1,314 @@
+
+Semantic Search
+Result: Under Joe Biden's leadership, America has deepened its relationship with NATO by responding to global challenges and strengthening NATO, making it stronger and more united than it has ever been, as we look to continue promoting stability, prosperity, and growth.
+
+Hybrid Search
+Result: Under Joe Biden's leadership, the United States has worked to deepen NATO's relationship through various means. First, he has emphasized the importance of NATO's collective defense commitment, reiterating that the United States will defend every inch of NATO territory, which has helped build trust and confidence among alliance members. Second, Biden has played a key role in modernizing NATO's strategies and capabilities to address emerging threats, such as cyber-attacks and hybrid warfare. Third, he has prioritized engagement with key NATO partners, such as Sweden and Finland, to broaden and deepen the transatlantic bond. Fourth, Biden has advocated for increased funding and resources for NATO, which has helped strengthen the alliance's military and economic capabilities. Overall, Biden's efforts have contributed to a more cohesive and effective NATO, which is better equipped to address global challenges and opportunities.
+
+```
+The results from the hybrid search appear to be slightly more detailed. They provide a more comprehensive response, including the names of countries and other specific information.
+Let’s add specific locations to see how both approaches fare.
+
+```
+Query: How is  Strait of Hormuz important in Biden's strategy?
+
+Semantic Search
+Result:  The passage doesn't explicitly state how the Strait of Hormuz is important in Biden's strategy. However, it does mention that the United States and its allies have a shared responsibility to maintain a free and open Indo-Pacific, which includes protecting the freedom of navigation. It is possible that preserving freedom of navigation in this area is also a part of Biden's overall strategy for the region.
+
+Hybrid Search
+Result: Biden's strategy emphasizes the importance of ensuring freedom of navigation through the Strait of Hormuz, a waterway located in the Middle East. This strategy aims to prevent any country from dominating the region through military efforts and ensures that there are no efforts to control the waterways. This emphasis on freedom of navigation is crucial for the United States and its allies as a significant portion of the world's oil supply passes through this waterway. Any disruption or control of this waterway could have significant economic and geopolitical implications, making Biden's strategy to maintain this freedom critical.
+
+```
+
+Again hybrid search fares well in location filters. Thus we can conclude that semantic search is enough for all common RAG systems but when it comes to specific names, filters, or abbreviations like in research papers, you may have to consider hybrid search.

```

Comment:
I think this conclusion is sound, but based on false pretenses.
The reason you need hybrid search is that keyword based search (which is also vector search based by the way, as you stated before) is implemented differently in the packages that you use and you need hybrid search to incorporate keyword based matching.

I think in general you should have less overarching statements as a lot of what you say is not theoretically true, but specific to these tools that you use.

Also justification for hybrid search in theory should come from somewhere else. There are considerations that you can implement using vector search (and that is preferable because of the drawbacks you mention for hybrid search), and there are some which are tricky or even impossible to do so (like hard 100% effective filters are close to impossible to implement in a vector similarity manner), and for those you might need hybrid search.